### PR TITLE
[BACKLOG-2051]: Implement support SSL for Carte

### PR DESCRIPTION
**Pull request by TatsianaKasiankova**
There are implementation and unit tests.

### CODE REVIEW

**Comment by TatsianaKasiankova** [comment]
@deinspanjer  Hi, Daniel
Please take a look this one.

File: engine/src/org/pentaho/di/www/WebServer.java [hunk 1]
```diff
@@ -77,9 +78,11 @@
 
   private String passwordFile;
 
+  private SslConfiguration sslConfig;
+
   public WebServer( LogChannelInterface log, TransformationMap transformationMap, JobMap jobMap,
       SocketRepository socketRepository, List<SlaveServerDetection> detections, String hostname, int port,
-      boolean join, String passwordFile ) throws Exception {
+      boolean join, String passwordFile, SslConfiguration sslConfig ) throws Exception {
```
Comment (mattyb149) [inline engine/src/org/pentaho/di/www/WebServer.java]: As a public "API" (constructor), should we keep the original as deprecated and have it call a new constructor with a null SSL config?

Comment (pentaho-nbaker) [inline engine/src/org/pentaho/di/www/WebServer.java]: Agree with Matty

**Comment by pentaho-nbaker** [comment]
Aliaksandr_Drebenchuk is going to add back the original constructor in the next commit

**Pull request merged**